### PR TITLE
Optimization for Ethereum signature

### PR DESCRIPTION
**Pull request by matteo-cristino**

### CODE REVIEW

**Comment by jaromil** [comment]
and once again well done sir! awaiting @albertolerda to merge

**Comment by albertolerda** [comment]
Yes, we can merge, I was wondering if it was worth importing shr from milagro, instead of using ddiv

**Comment by jaromil** [comment]
Indeed shift right by 1 should optimize a lot

File: src/zen_big.c [hunk 1]
```diff
@@ -730,7 +730,7 @@ static int big_mul(lua_State *L) {
 static int big_modpower(lua_State *L) {
 	big *x = big_arg(L,1); SAFE(x);
 	big *n = big_arg(L,2); SAFE(n);
-	big *m = big_arg(L,3); SAFE(n);
+	big *m = big_arg(L,3); SAFE(m);
```
Comment (jaromil) [inline src/zen_big.c:733]: well spotted! thanks

**Comment by jaromil** [comment]
well done implementing and overloading `__shr`!

**Pull request merged**